### ISSUE TITLE
TLS Encryption - Failed to start OvenMediaEngine

### ISSUE THREAD

**Issue by marijnv2000** [AirenSoft/OvenMediaEngine#903]
**PROBLEM**
Failed to start OvenMediaEngine after configuring [TLS Encryption](https://airensoft.gitbook.io/ovenmediaengine/configuration/tls-encryption). OvenMediaEngine starts fine with the `<Managers></Managers>` block commented out (no error when starting the service).

Certbot provided me with the following files under `/etc/letsencrypt/live/stream.vroem.xyz/`:
- cert.pem
- chain.pem
- fullchain.pem
- privkey.pem

I've tried different combinations of these, but I assume the config is expecting cert.pem, privkey.pem, chain.pem respectively.
I've also tried using the files available under `/etc/letsencrypt/archive/stream.vroem.xyz` with no success.

I'm unsure if I'm providing the wrong format or if it's a permission issue. Permissions on the directory and files are as follows:
```
ubuntu@ovenmediaengine:~$ ls -la /etc/letsencrypt/live/stream.vroem.xyz/
total 24
drwxr-xr-x 2 root root 4096 Sep 30 18:06 .
drwxr-xr-x 3 root root 4096 Sep 29 23:46 ..
-rw-r--r-- 1 root root  692 Sep 29 23:46 README
lrwxrwxrwx 1 root root   40 Sep 29 23:46 cert.pem -> ../../archive/stream.vroem.xyz/cert1.pem
lrwxrwxrwx 1 root root   41 Sep 29 23:46 chain.pem -> ../../archive/stream.vroem.xyz/chain1.pem
lrwxrwxrwx 1 root root   45 Sep 29 23:46 fullchain.pem -> ../../archive/stream.vroem.xyz/fullchain1.pem
lrwxrwxrwx 1 root root   43 Sep 29 23:46 privkey.pem -> ../../archive/stream.vroem.xyz/privkey1.pem
```

Permissions are adjusted from Certbot defaults as per [Certbot docs](https://eff-certbot.readthedocs.io/en/stable/using.html#where-are-my-certificates):
- Set 0755 on `/etc/letsencrypt/live` and `/etc/letsencrypt/archive` directories
- Set 0640 on `/etc/letsencrypt/live/stream.vroem.xyz/privkey.pem`
- Dit not run `chgrp` as I couldn't find out what the OvenMediaEngine service was running as.

The error in the ovenmediaengine and journalctl logs don't point me to a specific config line or entry.

**EXPECTATION**
I need some guidance to get the service to start with the TLS Encryption configuration in place

**ENVIRONMENTAL INFORMATION**
- OvenMediaEngine 0.14.10
- Ubuntu 22.04
- Used 0.14.9 prerequisites script and built locally as per [Getting Started](https://airensoft.gitbook.io/ovenmediaengine/getting-started#manual-installation-and-execution)

**SETUP INFORMATION AND LOGS**
[Server.xml.txt](https://github.com/AirenSoft/OvenMediaEngine/files/9686900/Server.xml.txt)
[ovenmediaengine.log](https://github.com/AirenSoft/OvenMediaEngine/files/9686901/ovenmediaengine.log)

**OTHER HELPFUL INFORMATION**

I found
- #198 - Suggests PEM format is fine (more recent)
- #144 - Suggests PEM needs to be converted

**Comment by bchah** [comment]
Your OME log shows two things - there are some unrecognized tags in your XML and it doesn't appear to be actually looking for your certificates. Glancing at the Server.xml you provided, you have a <Host> listed under <Managers> but that info should be under the <VirtualHosts> instead, where the template info still appears to be.

As a best practice I recommend going through the documentation and building out your own Server.xml with just the modules and settings you need - it will solve the above issues and prevent unneeded use of system resources.

Here's the section(s) I'm talking about:

```
..........<Managers>
		<Host>
			<Names>
				<Name>stream.vroem.xyz</Name>
			</Names>
			<TLS>
				<CertPath>/etc/letsencrypt/archive/stream.vroem.xyz/fullchain1.pem</CertPath>
				<KeyPath>/etc/letsencrypt/archive/stream.vroem.xyz/privkey1.pem</KeyPath>
				<ChainCertPath>/etc/letsencrypt/archive/stream.vroem.xyz/chain1.pem</ChainCertPath>
			</TLS>
		</Host>
		<API>
			<AccessToken>ome-access-token</AccessToken>

			<CrossDomains>
				<Url>*.airensoft.com</Url>
				<Url>http://*.sub-domain.airensoft.com</Url>
				<Url>http?://airensoft.*</Url>
			</CrossDomains>
		</API>
	</Managers>
	-->

	<VirtualHosts>
		<!-- You can use wildcard like this to include multiple XMLs -->
		<VirtualHost include="VHost*.xml" />
		<VirtualHost>
			<Name>default</Name>
			<!--Distribution is a value that can be used when grouping the same vhost distributed across multiple servers. This value is output to the events log, so you can use it to aggregate statistics. -->
			<Distribution>ovenmediaengine.com</Distribution>

			<!-- Settings for multi ip/domain and TLS -->
			<Host>
				<Names>
					<!-- Host names
						<Name>stream1.airensoft.com</Name>
						<Name>stream2.airensoft.com</Name>
						<Name>*.sub.airensoft.com</Name>
						<Name>192.168.0.1</Name>
					-->
					<Name>*</Name>
				</Names>
				<!--
				<TLS>
					<CertPath>path/to/file.crt</CertPath>
					<KeyPath>path/to/file.key</KeyPath>
					<ChainCertPath>path/to/file.crt</ChainCertPath>
				</TLS>
				-->
			</Host>......

**Comment by marijnv2000** [comment]
I'm an idiot. Thank you very much for pointing that out. I just found the first entry in the config file matching what was displayed on the [TLS Encryption](https://airensoft.gitbook.io/ovenmediaengine/configuration/tls-encryption) page and went from there. Clearly I should have paid more attention.

The service is starting now that I commented out the `Managers` block and moved the configuration to the `VirtualHosts` block. I will take your advice and go through the documentation to set the rest of my config properly.

Thanks again.
